src/pages/LoginPage.js

```diff
@@ -55,15 +55,31 @@ const LoginPage = ({ logo }) => {
   const onSubmit = async (e) => {
     e.preventDefault();
     setAuthenticating(true);
-    const response = await auth.login(credentials);
-    const { loginStatus, message } = response;
-    setServerResponse({ loginStatus, message });
-    if (response?.loginStatus !== "CORE_AUTH_ERR") {
-      history.push("/");
-    } else {
+  
+    try {
+      const response = await auth.login(credentials);
+      if (response.payload?.errors?.length) {
+        handleLoginError(response.payload.errors[0].message);
+        return;
+      }
+  
+      const { loginStatus, message } = response;
+      setServerResponse({ loginStatus, message });
+  
+      if (loginStatus === "CORE_AUTH_ERR") {
+        setAuthenticating(false);
+      } else {
+        history.push("/");
+      }
+    } catch (error) {
       setAuthenticating(false);
     }
   };
+  
+  const handleLoginError = (errorMessage) => {
+    setServerResponse({ loginStatus: "CORE_AUTH_ERR", message: errorMessage });
+    setAuthenticating(false);
+  };
```

Comment:
It's generally a good practice to declare functions before they are used (avoid issues with hoisting, makes code more readable). Could you declare `handleLoginError` before `onSubmit`?